src/device-api/actions.ts

```diff
@@ -279,15 +265,35 @@ export const executeServiceAction = async ({
 		throw new NotFoundError(messages.targetServiceNotFound);
 	}
 
-	// Execute action on service
-	return await executeDeviceAction(
-		generateStep(action, {
-			current: currentService,
-			target: targetService,
-			wait: true,
-		}),
-		force,
-	);
+	try {
+		// Take lock for service
+		await executeDeviceAction(
```

Comment:
I'm not sure this should be here. We don't need to take lock on `start`, for instance. The method taking calling `executeServiceAction` should take the lock. Alternatively you could have a wrapping function `executeServiceActionWithLock` that takes the lock before calling `executeServiceAction`

Reply:
What happens if the lock is taken here BTW? Will `executeDeviceActiond` throw? What error will be reported to the user then?

Reply:
If the lock is taken, the endpoint would return 423 Locked and `{"status":"failed","message":"Lockfile exists for { appId: $APP_ID, service: $SERVICE_NAME }"` due to the endpoint error handler which catches all 423's

Reply:
Added lock wrapping function: https://github.com/balena-os/balena-supervisor/pull/2234/commits/c1082b6b90aa31ba256cac2f43a80b24340b25cb#diff-d953a5d18d116d746cd0b200a5d7e917714811bc4609e71fc19196941bcd7ef0R218

Reply:
I don't think I expressed well on my comment, but the result is not what I meant. The `executeServiceAction` function is still taking a lock on `start` actions. I'll add some more comments on the commit